plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -59,16 +65,63 @@ var _ kubeapiserveradmission.WantsCloudConfig = &persistentVolumeLabel{}
 // As a side effect, the cloud provider may block invalid or non-existent volumes.
 func NewPersistentVolumeLabel() *persistentVolumeLabel {
 	return &persistentVolumeLabel{
-		Handler: admission.NewHandler(admission.Create),
+		Handler: admission.NewHandler(admission.Create, admission.Delete),
 	}
 }
 
 func (l *persistentVolumeLabel) SetCloudConfig(cloudConfig []byte) {
 	l.cloudConfig = cloudConfig
 }
 
+var _ = kubeapiserveradmission.WantsInternalKubeInformerFactory(&persistentVolumeLabel{})
+var _ = kubeapiserveradmission.WantsInternalKubeClientSet(&persistentVolumeLabel{})
+
+func (l *persistentVolumeLabel) SetInternalKubeClientSet(client internalclientset.Interface) {
+	l.client = client
+}
+
+func (l *persistentVolumeLabel) SetInternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	podInformer := f.Core().InternalVersion().Pods()
+	l.SetReadyFunc(podInformer.Informer().HasSynced)
+	l.podLister = podInformer.Lister()
+}
+
+func (l *persistentVolumeLabel) Validate() error {
+	if l.podLister == nil {
+		return fmt.Errorf("missing podLister")
+	}
+	if l.client == nil {
+		return fmt.Errorf("missing client")
+	}
+	return nil
+}
+
 func (l *persistentVolumeLabel) Admit(a admission.Attributes) (err error) {
-	if a.GetResource().GroupResource() != api.Resource("persistentvolumes") {
+	glog.V(1).Infof("IDC - Admit")
```

Comment:
Can you clean up all of these debugging messages?  A few are ok to keep around, but update the log level.